src/platform/plugins/shared/dashboard/public/dashboard_api/performance/query_performance_tracking.ts

```diff
@@ -123,8 +131,52 @@ function reportPerformanceMetrics({
 }) {
   const duration =
     loadType === 'dashboardSubsequentLoad' ? timeToData : Math.max(timeToData, totalLoadTime);
+  const groupedPerformanceMarkers = getPerformanceTrackersGroupedById(
```

Comment:
Thanks for putting together lens metrics. These will be really useful

Our goal for dashboard application is to be agnostic about embeddable types and implementations. Placing specific "lens" logic into dashboard plugin goes against this principle.

Dashboard uses interfaces to decouple embeddable specific knowledge. Would it be possible to implement this with an interface? For example, an `PerformanceTracker` interface.

Then, instead of calling out lens specifically, the logic could just group embeddable types that implement this interface?

Reply:
thx @markov00 for summarizing our discussion.

I think we want to strike a balance here between getting immediate value and increasing the footprint of the telemetry. Having some additional information about the time distributions of how panels render on an dashboard is an important gap that needs closing. This PR adds a high level metric for that, which will add color to the baseline.

To avoid tying this to "Lens", agreed that renaming it to "panel" is a good compromise.

Reply:
Thanks for the feedback everyone! I pushed some updates so all references to "Lens" now refer to "Panel".